tutor/src/components/change-student-id.cjsx

```diff
@@ -0,0 +1,100 @@
+_ = require 'lodash'
+React = require 'react'
+BS = require 'react-bootstrap'
+
+LoadableItem = require './loadable-item'
+{ ChangeStudentIdForm } = require 'shared'
+{ TransitionActions, TransitionStore } = require '../flux/transition'
+{ CourseStore, CourseActions } = require '../flux/course'
+{ StudentIdStore, StudentIdActions, ERROR_MAP } = require '../flux/student-id'
+Icon = require './icon'
+
+module.exports = React.createClass
+  contextTypes:
+    router: React.PropTypes.func
+
+  onCancel: -> @goBack()
+
+  componentWillMount: ->
+    {courseId} = @context.router.getCurrentParams()
+    CourseStore.once('course.loaded', @update)
+    CourseActions.load(courseId)
```

Comment:
From what I can see, there is nothing that will get the `CourseStore` to reload it's student id once `student-id-saved`  fires.  It looks like it will only get updated if the student comes back to update their id again.